File: PE4.py
#Cycle through the potential numbers, from largest (1MM) to smallest (10,000) 
#For each of those numbers, check to see 
	#(1) Check to see if the number is a palindrome. If yes, then:
	#(2) Check to see if it's a product of two 3-digit numbers
	

switch = 0

for i in range(999999,100000,-1):
	s = str(i)
	if s[0] == s[-1] and s[1] == s[-2] and s[2] == s[-3] and switch == 0:
		for j in range(999, 100, -1):
			if i % j == 0:
				t = str(i/j)
				if len(t) == 3:
					print(i)
					switch = 1 
					break
	

File: PE4.rb
#Write a method that tests a number for palindromocity
#Write a script that loops through products of two 3-digit numbers

def palindrome?(number)
  number = number.to_s
  number_middle_pt = ((number.length) / 2).round - 1
  (0..number_middle_pt).each do |index|
    reverse_index = -1*(index+1)
    if number[index] != number[reverse_index]
      puts "false"
      return false
    end
  end
  puts "true"
  return true
end

999.downto(100) do |num1|
  (999).downto(900) do |num2|
    product = num1*num2
    if palindrome?(product)
      puts product
      return product
    end
  end
end